aea/configurations/base.py

```diff
@@ -455,6 +455,17 @@ def to_latest(self) -> "PublicId":
         """Return the same public id, but with latest version."""
         return PublicId(self.author, self.name, self.LATEST_VERSION)
 
+    @classmethod
+    def is_valid_str(cls, public_id_string: str) -> bool:
+        """
+        Check if a string is a public id.
+
+        :param public_id_string: the public id in string format.
+        :return: bool indicating validity
+        """
+        match = re.match(cls.PUBLIC_ID_REGEX, public_id_string)
+        return match is not None
+
     @classmethod
     def from_str(cls, public_id_string: str) -> "PublicId":
```

Comment:
why not to use `is_valid_str` in `from_str` method?

Reply:
is_valid_str only returns bool, from_str needs match